src/drivers/indexeddb.js

```diff
@@ -345,6 +345,38 @@ function _tryReconnect(dbInfo) {
         });
 }
 
+// Safari could garbage collect transaction before oncomplete/onerror/onabord being dispatched
+// reference transaction to stop it being garbage collected and remove the reference when it finish
+var _refTransaction = {};
+var _refTransactionId = 0;
+
+function createReadWriteTransactionHelper() {
+    var unref = undefined;
+    return {
+        create: function(dbInfo, callback, retries) {
+            createTransaction(
+                dbInfo,
+                READ_WRITE,
+                function(err, transaction) {
+                    var id = _refTransactionId++;
+                    _refTransaction[id] = transaction;
```

Comment:
Technically very unlikely to happen, but it would be safer to `% Number.MAX_SAFE_INTEGER` the `_refTransactionId` for applications which run for a long time.